src-web/reducers/BaseServiceReducer.js

```diff
@@ -34,7 +34,7 @@ const _selectedNamespace = getSelectedNamespace()
 
 const initState = {
     namespaces: [],
-    secrets: [],
+    secrets: undefined,
```

Comment:
The motivation is to avoid fetching secrets (i.e. API call) from k8 as much as possible.  The rational for changing this from an empty array to `undefined` is because there are times where an empty array is a valid response from fetching secrets.  This could lead to having the page constantly fetching for secrets, when it's already known to have no secrets (i.e. empty array).  With `undefined`, we can know clearly when fetch has been done or not.